rbx_types/src/attributes/writer.rs

```diff
@@ -0,0 +1,136 @@
+use super::*;
+
+use crate::{basic_types::*, variant::Variant};
+
+use std::{borrow::Borrow, io::Write};
+
+fn write_f32<W: Write>(mut writer: W, n: f32) {
+    writer
+        .write_all(&n.to_le_bytes().to_vec())
+        .expect("couldn't write f32 to buffer");
```

Comment:
We should not panic, we should propagate this error.

Reply:
It was a panic because it's using a vector buffer, but now that it's going to be a Write, propagating makes more sense.